packages/widget/src/stores/form/useFieldActions.ts

```diff
@@ -16,5 +56,43 @@ export const useFieldActions = () => {
     shallow,
   );
 
-  return actions;
+  const setFieldValueWithEmittedEvents = useCallback(
+    (
+      fieldName: FormFieldNames,
+      value: GenericFormValue,
+      options?: SetOptions,
+    ) => {
+      emitEventForFieldValueChange(
+        fieldName,
+        value,
+        actions.getFieldValues(fieldName)[0],
+        emitter,
+      );
+
+      actions.setFieldValue(fieldName, value, options);
+    },
+    [actions, emitter],
+  );
+
+  const setUserAndDefaultValuesWithEmittedEvents = useCallback(
+    (formValues: Partial<DefaultValues>) => {
+      (Object.keys(formValues) as FormFieldNames[]).forEach((fieldName) => {
+        emitEventForFieldValueChange(
+          fieldName,
+          formValues[fieldName],
+          actions.getFieldValues(fieldName)[0],
+          emitter,
+        );
+      });
+
+      actions.setUserAndDefaultValues(formValues);
```

Comment:
Should we set value first and, after that fire an event?

Reply:
The actions.getFieldValues will not have the old value to do the comparison to check for change if you call actions.setFieldValue first. I tried it the other way around initially

Reply:
I guess we can save it to a variable like 

`const oldValue = actions.getFieldValues(fieldName)[0]`

and then execute after.

Reply:
That would actually become a bit more complicated in the example above for the `setUserAndDefaultValuesWithEmittedEvents` - you would have to store the oldValues in another object map first and then once `setUserAndDefaultValues` is executed you would have to the iterate over the old values map.

I don't see what the benefit will be of changing the order of execution there and I think it will make the code a little clean - but I do it and can decide if you are happier with than

Reply:
Have put something in place for this